source/styleguide/components/SgColorSwatch/SgColorSwatch.jsx

```diff
@@ -0,0 +1,154 @@
+import SgHeading from '@sg-tags/SgHeading/SgHeading';
+import { createObject, getContrast, runWCAGTest } from './SgColorSwatch.Container';
+
+/* variables file generated by the postcss-export-vars plugin. */
+import colorVars from './SgColorSwatch__variables.json';
+
+export const SgColorSwatch = (props) => {
+  const {
+    tagName: Tag,
+    className,
+    children,
+    ...attrs
+  } = props;
+
+  const classStack = FcUtils.createClassStack([
+    'SgColorSwatch',
+    className
+  ]);
+
+  return (
+    <SgColorSwatch__wrapper>
+      {
+        colorVars ?
+        Object.keys(colorVars).map((color, i) => {
+          const title = color.substr(color.indexOf('color') + 5).split(/(?=[A-Z])/).join(' ');
+          const obj = createObject(colorVars[color]);
+
+          /* contrast tests */
+          const contrastPrimary = getContrast(obj.hex, colorVars.colorTextPrimary);
+          const contrastSecondary = getContrast(obj.hex, colorVars.colorTextSecondary);
+
+          return (
+            <Tag
+              key={i}
+              className={classStack}
+              {...attrs}
+              style={{ backgroundColor: obj.hex }}
+            >
+              <SgColorSwatch__accessibility variant="no-badge" contrastPrimary={contrastPrimary} contrastSecondary={contrastSecondary} level="AA" />
+              <SgColorSwatch__accessibility variant="no-badge" contrastPrimary={contrastPrimary} contrastSecondary={contrastSecondary} level="AAA" />
+
+              <SgColorSwatch__panel>
```

Comment:
Knowing how this compiles down, this might not be possible; but do we find value in knowing where all these color comes from? (styleguide/variables/colors.css vs. elements/variables/colors.css)

Reply:
Ideally the ColorSwatches only show project colors (from the `elements/;variables/colors.css` file). When building a project, theoretically no dev should be using the colors from the Styleguide.